packages/actor-query-operation-leftjoin/test/ActorQueryOperationLeftJoin-test.ts

```diff
@@ -124,6 +124,82 @@ describe('ActorQueryOperationLeftJoin', () => {
       });
     });
 
+    it('should correctly handle left hand bindings that are missing the variables of the expression', async() => {
```

Comment:
I don't fully understand this test.
Won't this test also pass on the old code?

I would have expected a distinction between the two calls to `mediatorQueryOperation`, and return different metadata and bindings.
I also would expect an actual expression, which is only evaluated when bindings variables are complete, in which case we should check for the expected expression result.

Reply:
I have written this test to get 100% code coverage.
It checks the code path where there are no variables.
Do you see a better way to check for this code path?

Reply:
One way of doing this would be to check the `arg` in `mediatorQueryOperation.mediate`. If it's the left stream (need something to identify this when passing it to `op`), then you return one thing (bindingsStream+metadata+...). If it's the right stream, you return something else.

Reply:
If I did understand it correctly, the test checks:

- a binding from the left stream
- that does not have the variables from the expression
- so it must be push without being evaluated

I have removed the cruft around the test so now it only checks if it indeed has the binding that did pass through the if statement that checks if it does not contains all the variables.

@rubensworks do you think it is now okay?

Reply:
I mean something like this:
```
mediatorQueryOperation.mediate = (arg: any) => {
  if (arg is of left stream) return x;
  else return y;
}
```

Reply:
Did you see my last comment above?

Reply:
Hi @rubensworks 

Thanks for the heads up, now I have seen it.
I will have a look at it. 

I can not yet piece it together, I feel like the test is already passing through the code.

What would it improve? if we return a different thing depending on the arg?

Reply:
> What would it improve?

Code clarity.

Indeed, the test is already causing full coverage of the code. But this feels very arbitrary when just reading the test.
By explicitly adding the if-statement in the test, it should become clearer to the code reader what the test is doing exactly.

Reply:
okay, thanks, I will implement this